integration/src/integration/java/pl/allegro/tech/hermes/integration/management/QueryEndpointTest.java

```diff
@@ -0,0 +1,115 @@
+package pl.allegro.tech.hermes.integration.management;
+
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+import pl.allegro.tech.hermes.api.EndpointAddress;
+import pl.allegro.tech.hermes.api.Group;
+import pl.allegro.tech.hermes.api.Subscription;
+import pl.allegro.tech.hermes.api.Topic;
+import pl.allegro.tech.hermes.integration.IntegrationTest;
+
+import java.util.List;
+
+import static pl.allegro.tech.hermes.api.ContentType.AVRO;
+import static pl.allegro.tech.hermes.api.ContentType.JSON;
+import static pl.allegro.tech.hermes.api.Subscription.Builder.subscription;
+import static pl.allegro.tech.hermes.api.SubscriptionPolicy.Builder.subscriptionPolicy;
+import static pl.allegro.tech.hermes.api.Topic.Builder.topic;
+import static pl.allegro.tech.hermes.integration.test.HermesAssertions.assertThat;
+
+public class QueryEndpointTest extends IntegrationTest {
+
+    @DataProvider(name = "groupData")
+    public static Object[][] groupData() {
+        return new Object[][] {
+                {4, "{\"query\": {}}"},
+                {1, "{\"query\": {\"groupName\": \"testGroup1\"}}"},
+                {1, "{\"query\": {\"groupName\": {\"like\": \".*Group2\"}}}"},
+                {3, "{\"query\": {\"groupName\": {\"like\": \".*Group.*\"}}}"},
+                {1, "{\"query\": {\"technicalOwner\": \"Owner2\", \"supportTeam\": \"Support3\"}}"},
+                {1, "{\"query\": {\"and\": [{\"technicalOwner\": \"Owner2\"}, {\"supportTeam\": \"Support3\"}]}}"},
+                {3, "{\"query\": {\"or\": [{\"technicalOwner\": \"Owner2\"}, {\"supportTeam\": \"Support3\"}]}}"},
+        };
+    }
+
+    @Test(dataProvider = "groupData")
+    public void shouldQueryGroup(Integer resultSize, String query) {
+        // given
+        management.group().create(new Group("testGroup1", "Owner1", "Support3", "Contact1"));
+        management.group().create(new Group("testNot1", "Owner2", "Support3", "Contact1"));
+        management.group().create(new Group("testGroup2", "Owner2", "Support4", "Contact2"));
+        management.group().create(new Group("testGroup3", "Owner1", "Support2", "Contact1"));
+
+        // when
+        List<Group> found = management.query().queryGroup(query);
+
+        // then
+        assertThat(found).hasSize(resultSize);
```

Comment:
This feels neat, but I'm reluctant to admit it feels correct. Can we also test whether the result matches specific 'ids' ?